pkgs/by-name/gr/grafana-alloy/package.nix

```diff
@@ -0,0 +1,116 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchYarnDeps
+, buildGoModule
+, systemd
+, yarn
+, fixup-yarn-lock
+, nodejs
+, grafana-alloy
+, nix-update-script
+, nixosTests
+, testers
+}:
+
+buildGoModule rec {
+  pname = "grafana-alloy";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "grafana";
+    repo = "alloy";
+    hash = "sha256-jaOQG+QkVi10lUp6izvSGE9j76ULc4yKbxdDhLHykYI=";
+  };
+
+  proxyVendor = true;
+  vendorHash = "sha256-6Xc2siImM1Dl716uGhtAGcn+PO2OLuYLxanzg8Ho6SA=";
+
+  nativeBuildInputs = [ fixup-yarn-lock yarn nodejs ];
+
+  ldflags =
+    let
+      prefix = "github.com/grafana/alloy/internal/build";
+    in
+    [
+      "-s"
+      "-w"
+      # https://github.com/grafana/alloy/blob/3201389252d2c011bee15ace0c9f4cdbcb978f9f/Makefile#L110
+      "-X ${prefix}.Branch=v${version}"
+      "-X ${prefix}.Version=${version}"
+      "-X ${prefix}.Revision=v${version}"
+      "-X ${prefix}.BuildUser=nix"
+      "-X ${prefix}.BuildDate=1980-01-01T00:00:00Z"
+    ];
+
+  tags = [
+    "nonetwork" # disable network tests
+    "nodocker" # disable docker tests
+    "netgo"
+    "builtinassets"
+    "promtail_journal_enabled"
+  ];
```

Comment:
`nonetwork` and `nodocker` are test-only tags and should be moved to `checkFlags`, as they would otherwise show up in `alloy --version`:

```shell
# before:
# result/bin/alloy --version
alloy, version v1.1.1 (branch: v1.1.1, revision: v1.1.1)
  build user:       nix
  build date:       1980-01-01T00:00:00Z
  go version:       go1.22.3
  platform:         linux/amd64
  tags:             nonetwork,nodocker,netgo,builtinassets,promtail_journal_enabled

# after:
# result/bin/alloy --version
alloy, version v1.1.1 (branch: v1.1.1, revision: v1.1.1)
  build user:       nix
  build date:       1980-01-01T00:00:00Z
  go version:       go1.22.3
  platform:         linux/amd64
  tags:             netgo,builtinassets,promtail_journal_enabled
```

```suggestion
  tags = [
    "netgo"
    "builtinassets"
    "promtail_journal_enabled"
  ];

  checkFlags = [
    "-tags nonetwork" # disable network tests
    "-tags nodocker" # disable docker tests
  ];
```

`nonetwork` and `nodocker` looks mostly like a leftover from `grafana-agent` tbh.
The checkPhase runs with and without those flags successfully and don't /actually/ require network nor docker (yet?).
But since upstream has them in their CI as well, and they don't do harm, we should probably just keep them.

Reply:
Agreed, but let's stay in sync with their CI here. I can probably send a PR upstream to remove it.

Reply:
i grepped the repo, there's still a bunch of build flags in various files:

```

internal/component/remote/vault/vault_test.go
1://go:build !nodocker

internal/static/integrations/cadvisor/cadvisor_test.go
1://go:build !nonetwork && !nodocker && linux

internal/tools/packaging_test/environment_test.go
1://go:build !nonetwork && !nodocker && !race && packaging
2:// +build !nonetwork,!nodocker,!race,packaging

internal/tools/packaging_test/alloy_linux_packages_test.go
1://go:build !nonetwork && !nodocker && !race && packaging
2:// +build !nonetwork,!nodocker,!race,packaging
```

so it's probably intentional?

Reply:
Yes, I know. They just don't to a lot in (and we don't set `packaging`).